lib/plugins/ray_trace.js

```diff
@@ -23,4 +24,29 @@ module.exports = (bot) => {
 
     return bot.world.raycast(eyePosition, viewDirection, maxDistance, matcher)
   }
+
+  bot.entityAtCursor = (maxDistance = 3.5) => {
+    const entities = Object.values(bot.entities)
+      .filter(entity => entity.type !== 'object' && entity.username !== bot.username && entity.position.distanceSquared(bot.entity.position) <= maxDistance ** 2)
+      .sort((entity1, entity2) => entity1.position.distanceSquared(bot.entity.position) - entity2.position.distanceSquared(bot.entity.position))
+
+    const dir = new Vec3(-Math.sin(bot.entity.yaw) * Math.cos(bot.entity.pitch), Math.sin(bot.entity.pitch), -Math.cos(bot.entity.yaw) * Math.cos(bot.entity.pitch))
+    const iterator = new RaycastIterator(bot.entity.position.offset(0, bot.entity.height, 0), dir.normalize(), maxDistance)
+
+    let pos = iterator.next()
+    while (pos) {
+      for (let i = 0; i < entities.length; i++) {
```

Comment:
This code is checking all entities in range at each step of the recast, this is a lot of redundant work.
A way to improve that is to do independent raycasts for blocks and entities and then simply compare the distance of the first entity and the fist block.

In pseudocode:
* Do a block raycast using `const b = bot.world.raycast(eyePosition, viewDirection, maxDistance)`
* Update maxDistance: `maxDistance = Math.min(maxDistance, eyePosition.distanceSquared(b.intersect))`
* Do your filtering on entities
* Test entity intersects

Note that the 2 last steps can be merged in 1 loop to avoid creating temporary arrays
Note2: sorting the entities is not enough to ensure the first entity is returned, because some entities can be larger than others. You have to test them all and return the nearest, which shouldn't be an issue if you do the other simplifications.